### PR TITLE
Admin badge and tooltip adjustments

### LINKED ISSUE #673: Admin Badge + Tooltip Adjustments
Currently, we show little badges + tooltips for JC Admins and Agency Admins. These show up in two places: the Records page and the Uploaded Files page (in Settings). We'd like to make a couple small tweaks (all screenshots are from Clackamas County Jail agency):

* For Agency Admins, remove the badge and tooltip in both places

* For JC Admins, keep the badge, but remove the tooltip on hover that says JC Admin (it's redundant, because we're currently showing the name as JC Admin too)

* When you hover over a row with multiple users, if one of the users is a JC Admin, don't show the little badge icon in the tooltip (I think it looks weird!)

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Adjustments to editors tooltips and badges

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #673 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by ivan-kishko** [comment]
@lilidworkin I think I get the idea right but if anything let me know! Thanks!)

**Review by mxosman — COMMENTED** [review]
Awesome, Ivan! I have a couple notes on this:

1. When I run this with staging as the backend and I'm in a Contributor role, I notice that the badges don't appear for editors named JC Admin. I think the intention was to keep the badge within the row for those named "JC Admin", and remove the badge that appears inside of the tooltip. Please let me know if I'm misunderstanding!
    * Screenshot 2023-05-26 at 5 01 27 PM
2. There's some weird behavior with where the additional editors tooltip is appearing that might've always been there, but it's really noticeable now - LMK if you'd prefer to handle this in a separate PR and I'll write up a task. 


   * Sometimes the tooltip appears too far to the right - maybe we favor having it drop down left aligned to the first name/left edge of that cell.
     * Screenshot 2023-05-26 at 4 50 27 PM
     * Screenshot 2023-05-26 at 5 07 22 PM

   * It would be nice if the tooltip width was contained - so if there are like 8 additional editors, it should wrap instead of extend in a single horizontal line.
      * Screenshot 2023-05-26 at 4 51 17 PM

File: publisher/src/components/primitives/index.tsx [hunk 1]
```diff
@@ -75,32 +54,20 @@ export const TeamMemberNameWithBadge: React.FC<{
...
     <TeamMemberNameContainer color={badgeColor}>
       <NameContainer>{name}</NameContainer>
-      {role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN && (
-        <StyledRecidivizAdmin id={badgeId} />
-      )}
-      {role === AgencyTeamMemberRole.AGENCY_ADMIN && (
-        <StyledAgencyAdmin id={badgeId} />
-      )}
+      {role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN &&
+        name !== "JC Admin" && <StyledRecidivizAdmin id={badgeId} />}
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/primitives/index.tsx:58]: good that after commit it is used once)

**Comment by ivan-kishko** [comment]
> Awesome, Ivan! I have a couple notes on this:
> 
> 1. When I run this with staging as the backend and I'm in a Contributor role, I notice that the badges don't appear for editors named JC Admin. I think the intention was to keep the badge within the row for those named "JC Admin", and remove the badge that appears inside of the tooltip. Please let me know if I'm misunderstanding!
>    
>    * Screenshot 2023-05-26 at 5 01 27 PM
> 2. There's some weird behavior with where the additional editors tooltip is appearing that might've always been there, but it's really noticeable now - LMK if you'd prefer to handle this in a separate PR and I'll write up a task.
>    
>    * Sometimes the tooltip appears too far to the right - maybe we favor having it drop down left aligned to the first name/left edge of that cell.
>      
>      * Screenshot 2023-05-26 at 4 50 27 PM
>      * Screenshot 2023-05-26 at 5 07 22 PM
>    * It would be nice if the tooltip width was contained - so if there are like 8 additional editors, it should wrap instead of extend in a single horizontal line.
>      
>      * Screenshot 2023-05-26 at 4 51 17 PM

@mxosman [13979f9](https://github.com/Recidiviz/justice-counts/pull/674/commits/13979f9d9ab08bf6986e172dc8faa65487d5b9f9)

File: publisher/src/components/primitives/index.tsx [hunk 2]
```diff
@@ -71,36 +50,29 @@ export const TeamMemberNameWithBadge: React.FC<{
...
+  const isJCAdminInsideTooltip = isInsideTooltip && name === "JC Admin";
+  return (
+    <>
+      <TeamMemberNameContainer color={badgeColor}>
+        <NameContainer>{`${name}${isLast ? "" : ","}`}</NameContainer>
+        {role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN &&
+          !isJCAdminInsideTooltip && <StyledRecidivizAdmin id={badgeId} />}
+      </TeamMemberNameContainer>
+      {role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN &&
+        name !== "JC Admin" && (
```
Comment (lilidworkin) [inline publisher/src/components/primitives/index.tsx:65]: This looks exactly right!

File: publisher/src/components/primitives/index.tsx [hunk 3]
```diff
@@ -71,36 +50,29 @@ export const TeamMemberNameWithBadge: React.FC<{
...
-);
+  isInsideTooltip?: boolean;
+  isLast?: boolean;
+}> = ({ name, role, badgeColor, badgeId, isInsideTooltip, isLast }) => {
+  const isJCAdminInsideTooltip = isInsideTooltip && name === "JC Admin";
+  return (
+    <>
+      <TeamMemberNameContainer color={badgeColor}>
+        <NameContainer>{`${name}${isLast ? "" : ","}`}</NameContainer>
+        {role === AgencyTeamMemberRole.JUSTICE_COUNTS_ADMIN &&
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/primitives/index.tsx:65]: @lilidworkin Sorry, are you sure badge is not showing up inside cell for JC admin? Can you maybe update branch? Cause I have a badge next to JC Admin, BUT I dont have rows with exactly "JC Admin & others" to check a case you screenshoted

[the remark this answers is not in the document]
Comment (lilidworkin) [reply publisher/src/components/primitives/index.tsx:65]: You're right @ivan-kishko , my mistake, I must not have pulled latest changes!

**Review by lilidworkin — APPROVED** [review]
Looks good!!

**Pull request merged**